### ISSUE TITLE
Can't change model to 4, migration says model <4> not installed

### ISSUE THREAD

**Issue by adi2k5** [matiasdelellis/facerecognition#318]
### Expected behaviour

Model 4 should be installed and picked as default with occ face:setup -m 4.

### Actual behaviour

Model 4 is theoretically installed, but previous model is still in use, occ face:setup -m 4 does nothing except showing info that model has been installed.

### Steps to reproduce
1. Check which model is currently in use with occ face:setup
2. Run occ face:setup -m 4 to install new model
3. Validate again which model is in use and try to migrate to new model

```
$ php occ face:setup
You must indicate the ID of the model to install
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Id | Enabled | Name        | Description                                                                                                                                              |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| 1  |         | Default     | Main model, using dlib defaults: mmod_human_face_detector.dat, shape_predictor_5_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat        |
| 2  | *       | DlibCnn68   | Alternative default model, using dlib: mmod_human_face_detector.dat, shape_predictor_68_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat |
| 3  |         | DlibHog     | DDlib HOG Model which needs lower requirements                                                                                                           |
| 4  |         | DlibCnnHog5 | Extends the main model, doing a face validation with the Hog detector                                                                                    |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
$ php occ face:setup -m 4
The model 4 (DlibCnnHog5) will be installed
Install model 4 (DlibCnnHog5) successfully done
The model 4 (DlibCnnHog5) was configured as default
$ php occ face:migrate -m 4 -u ****
The model <4> is not installed
$ php occ face:setup
You must indicate the ID of the model to install
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Id | Enabled | Name        | Description                                                                                                                                              |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| 1  |         | Default     | Main model, using dlib defaults: mmod_human_face_detector.dat, shape_predictor_5_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat        |
| 2  | *       | DlibCnn68   | Alternative default model, using dlib: mmod_human_face_detector.dat, shape_predictor_68_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat |
| 3  |         | DlibHog     | DDlib HOG Model which needs lower requirements                                                                                                           |
| 4  |         | DlibCnnHog5 | Extends the main model, doing a face validation with the Hog detector                                                                                    |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Change to any other model, ie. model 3 works perfectly fine:
```
$ php occ -vvv face:setup
You must indicate the ID of the model to install
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Id | Enabled | Name        | Description                                                                                                                                              |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| 1  |         | Default     | Main model, using dlib defaults: mmod_human_face_detector.dat, shape_predictor_5_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat        |
| 2  | *       | DlibCnn68   | Alternative default model, using dlib: mmod_human_face_detector.dat, shape_predictor_68_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat |
| 3  |         | DlibHog     | DDlib HOG Model which needs lower requirements                                                                                                           |
| 4  |         | DlibCnnHog5 | Extends the main model, doing a face validation with the Hog detector                                                                                    |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
$ php occ -vvv face:setup -m 3
The model 3 (DlibHog) will be installed
Install model 3 (DlibHog) successfully done
The model 3 (DlibHog) was configured as default
$ php occ -vvv face:setup
You must indicate the ID of the model to install
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Id | Enabled | Name        | Description                                                                                                                                              |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| 1  |         | Default     | Main model, using dlib defaults: mmod_human_face_detector.dat, shape_predictor_5_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat        |
| 2  |         | DlibCnn68   | Alternative default model, using dlib: mmod_human_face_detector.dat, shape_predictor_68_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat |
| 3  | *       | DlibHog     | DDlib HOG Model which needs lower requirements                                                                                                           |
| 4  |         | DlibCnnHog5 | Extends the main model, doing a face validation with the Hog detector                                                                                    |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
```

### Server configuration

* **Operating system:** Fedora

* **Pdlib version:** 1.0.2-1

* **How is DLib installed:** dnf install using copr copr:copr.fedorainfracloud.org:matias:dlib

* **How is PDlib installed:** dnf install using copr copr:copr.fedorainfracloud.org:matias:dlib

* **PHP version:** 7.3.20

* **Web server:** lighttpd

* **Database:** mariadb

* **Nextcloud version:** 19.0.1

### Client configuration

* **Browser:** n/a, issue on server itself

* **Operating system:** n/a, issue on server itself

#### Background task log with debug.
<details>
<summary>php occ -vvv face:setup -m 4</summary>

```         
The model 4 (DlibCnnHog5) will be installed
Install model 4 (DlibCnnHog5) successfully done
The model 4 (DlibCnnHog5) was configured as default
```
</details>

#### Web server error log
<details>
<summary>No errors in webserver</summary>
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>No errors logged in nextcloud</summary>
</details>

#### Browser log
n/a

**Comment by matiasdelellis** [comment]
Hi @adi2k5 

> php occ face:migrate -m 4 -u ****

Here the option -m is the model you want to migrate to the current one. So use 1, 2, or 3

**Comment by adi2k5** [comment]
@matiasdelellis ok, but if I want to migrate from model 2 to model 4, I'm not able to as model 4 is not being really installed in my case:

```
$ php occ face:setup
You must indicate the ID of the model to install
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Id | Enabled | Name        | Description                                                                                                                                              |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| 1  |         | Default     | Main model, using dlib defaults: mmod_human_face_detector.dat, shape_predictor_5_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat        |
| 2  |         | DlibCnn68   | Alternative default model, using dlib: mmod_human_face_detector.dat, shape_predictor_68_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat |
| 3  | *       | DlibHog     | DDlib HOG Model which needs lower requirements                                                                                                           |
| 4  |         | DlibCnnHog5 | Extends the main model, doing a face validation with the Hog detector                                                                                    |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
$ php occ  face:setup -m 4
The model 4 (DlibCnnHog5) will be installed
Install model 4 (DlibCnnHog5) successfully done
The model 4 (DlibCnnHog5) was configured as default
$ php occ face:setup
You must indicate the ID of the model to install
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Id | Enabled | Name        | Description                                                                                                                                              |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| 1  |         | Default     | Main model, using dlib defaults: mmod_human_face_detector.dat, shape_predictor_5_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat        |
| 2  |         | DlibCnn68   | Alternative default model, using dlib: mmod_human_face_detector.dat, shape_predictor_68_face_landmarks.dat and dlib_face_recognition_resnet_model_v1.dat |
| 3  | *       | DlibHog     | DDlib HOG Model which needs lower requirements                                                                                                           |
| 4  |         | DlibCnnHog5 | Extends the main model, doing a face validation with the Hog detector                                                                                    |
+----+---------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
```

changing between models 1, 2 or 3 works as expected, but changing to new model won't have any effect - DlibCnnHog5 is not set up as default in any case.

**Comment by matiasdelellis** [comment]
Hi @adi2k5 

Ohh .. You are right. :disappointed: 

I think I discovered the problem.. This model depends on model 1 and model 3. then both must be installed. If you never installed model 1, please install it and then try again..
If it's already installed, I'm still investigating.. :disappointed:

**Comment by adi2k5** [comment]
@matiasdelellis you're right, that was exactly the problem. After installing model 1, now 4 is set to default. I think it would be good to add this info about model 1 and 3 being installed to model 4 prereqs :) 
Thanks!

**Comment by matiasdelellis** [comment]
Great!.
Although there is a message to indicate it, I honestly had not tried it. :disappointed: 

I open it for other users' reference until I fix it correctly. :wink:

**Comment by matiasdelellis** [comment]
Fixed in #325

Thanks again!. :smile:
